tests/Phar/GlobResourceTest.php

```diff
@@ -0,0 +1,16 @@
+<?php declare(strict_types=1);
+
+namespace Rector\Tests\Phar;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Config\Resource\GlobResource;
+
+final class GlobResourceTest extends TestCase
+{
+    public function testGlobResource(): void
+    {
+        $resource = new GlobResource('phar://' . __DIR__ . '/Source/rector.phar/src/', '', true);
+
+        $this->assertNotCount(0, $resource);
+    }
+}
```

Comment:
Here it's needed to override `getContainerLoader()` in `AppKernel` with own `PharAwaretGlobFileLoader` class:

```php
final class RectorKernel extends Kernel
{

    /**
     * @param ContainerInterface|ContainerBuilder $container
     */
    protected function getContainerLoader(ContainerInterface $container): DelegatingLoader
    {
        $kernelFileLocator = new FileLocator($this);

        $loaderResolver = new LoaderResolver([
            // @todo - create this clsas
            return PharAwaretGlobFileLoader($container, $kernelFileLocator));
            // ...
        ]);

        return new DelegatingLoader($loaderResolver);
    }

}
```

As done in here https://github.com/symplify/packagebuilder#10-do-you-need-to-merge-parameters-in-yaml-files-instead-of-override

---

`PharAwaretGlobFileLoader` might look like this:


```php
namespace Symfony\Component\Config\Loader\GlobFileLoader;

final class PharAwaretGlobFileLoader extends GlobFileLoader
{
    protected function glob(string $pattern, bool $recursive, &$resource = null, bool $ignoreErrors = false)
    {
        // override parent glob() method logic to not-use "glob" function
    }
}

Reply:
What do you think @mssimi ?

Reply:
Check GlobResource usages, I think it is not only used in loader.